packages/itwin/tree-widget/src/components/trees/category-tree/CategoriesTree.tsx

```diff
@@ -93,52 +97,64 @@ export function CategoryTree(props: CategoryTreeProps) {
       levelOffset: 10,
     },
   });
-  const overlay = isFiltering ? <div className="filteredTreeOverlay" /> : undefined;
-  const filterApplied = filteredNodeLoader !== nodeLoader;
-
   const noFilteredDataRenderer = useCallback(() => {
     return <VisibilityTreeNoFilteredData
       title={TreeWidget.translate("categoriesTree.noCategoryFound")}
       message={TreeWidget.translate("categoriesTree.noMatchingCategoryNames")}
     />;
   }, []);
 
+  if (!state) {
+    return null;
+  }
```

Comment:
When looking at this, I wanted to check how long it may take us until we get a non-undefined `state`. From the API consumer's perspective that's not clear - docs don't say anything about the `undefined` result neither on the `useVisibilityTree`, nor the `usePresentationTreeState`. I think they should both mention what `undefined` result means.